src/rimage.c

```diff
@@ -163,15 +168,21 @@ int main(int argc, char *argv[])
 		return resign_image(&image);
 	}
 
-	/* set IMR Type in found machine definition */
-	if (image.adsp->man_v1_8)
+	/* set IMR Type and production bit in found machine definition */
```

Comment:
one more field is required to properly setup imr_type, take a look here:

https://github.com/thesofproject/rimage/pull/113#discussion_r1001520828

Reply:
@abonislawski sorry, don't understand. It does say that the "-i" option already exists. What has to be done?

Reply:
set bitmap[4] to 0x10 if imr type = 4

Right now we have -i option but imr_type is not set properly, bitmap[4] = always 8 in current code